### PR TITLE
Feature/rework job scheduling

### PR DESCRIPTION
**Pull request by denis-yuen**
This seems to properly hook up DropWizard+Hibernate to create DB records compatible with our legacy daemon utilities. 

This properly schedules and lists workflows (without any auth).

Tested by doing POST with the following (proper testing follows in a different ticket) 

```
{
  "end_user": "Player1",
  "flavour": "m1.xlarge",
  "job_hash": "test",
  "extra_files":  { "/etc/init.d/something_important" : "jkajkggk
wegnkljkorgjklrgnklerklrg", "field2" : "value2"},
  "container_image_descriptor": "{ \"field1\" : \"value1\", \"field2\" :
\"value2\"}",
  "container_runtime_descriptor": "{ \"field1\" : \"value1\", \"field2\"
: \"value2\"}"
}
```

This also adds the auto-generated swagger Java client for us to build upon with no extra dependencies for the CLI and moves stuff into a consistent io.consonance namespace.

### CODE REVIEW

File: consonance-arch/sql/schema.sql [hunk 1]
```diff
@@ -1,61 +1,243 @@
...
+--
+
+CREATE TABLE job (
+    job_id integer NOT NULL,
+    container_image_descriptor text,
+    container_runtime_descriptor text,
+    create_timestamp timestamp without time zone,
+    end_user text,
+    flavour text,
+    ini text,
```
Comment (SolomonShorser-OICR) [inline consonance-arch/sql/schema.sql:69]: There's a new `ini_params` table, which I guess replaces this, but I'm also guessing that this field is still here for backwards compatibility?

File: consonance-arch/src/test/java/io/consonance/arch/coordinator/CoordinatorIT.java [hunk 2]
```diff
@@ -57,10 +58,10 @@ public void testNormalUsage() throws Exception {
         File file = FileUtils.getFile("src", "test", "resources", "config");
         File iniDir = FileUtils.getFile("ini");
         // prime the pump with a job
-        JobGenerator.main(new String[] { "--config", file.getAbsolutePath(), "--ini", iniDir.getAbsolutePath(), "--workflow-name",
-                "DEWrapper", "--workflow-version", "1.0.0", "--workflow-path",
-                "/workflows/Workflow_Bundle_DEWrapperWorkflow_1.0.0_SeqWare_1.1.0" ,
-                "--flavour","m1.xlarge"});
+        JobGenerator
+                .main(new String[] { "--config", file.getAbsolutePath(), "--ini", iniDir.getAbsolutePath(), "--workflow-name", "DEWrapper",
```
Comment (SolomonShorser-OICR) [inline consonance-arch/src/test/java/io/consonance/arch/coordinator/CoordinatorIT.java]: Why is DEWrapper still hardcoded in here? Should that be changed to HelloWorld?

File: consonance-arch/sql/schema.sql (hunk 1 above)
Comment (denis-yuen) [inline consonance-arch/sql/schema.sql:69]: Yep, once I get the new web client and worker, the old ini gets removed. 
But until that gets hooked up, I'll just leave the old versions for testing purposes.

File: consonance-common/src/main/java/io/consonance/arch/Base.java [hunk 3]
```diff
@@ -9,8 +8,11 @@
...
 import org.slf4j.LoggerFactory;
 
+import java.io.IOException;
+
 /**
- * Created by boconnor on 15-04-18.
+ * @author boconnor
+ * @author dyuen
  */
 public class Base {
```
Comment (SolomonShorser-OICR) [inline consonance-common/src/main/java/io/consonance/arch/Base.java:17]: This class is still in use?

File: consonance-common/src/main/java/io/consonance/arch/persistence/PostgreSQL.java [hunk 4]
```diff
@@ -187,11 +188,10 @@ public Integer createProvision(Provision p) {
 
     public String createJob(Job j) {
         JSONObject jsonIni = new JSONObject(j.getIni());
-        JSONObject jsonExtraFiles = new JSONObject(j.getExtraFiles());
         Map<Object, Map<String, Object>> map = this.runInsertStatement(
-                "INSERT INTO job (status, job_uuid, workflow, workflow_version, job_hash, ini, extra_files) VALUES (?,?,?,?,?,?,?::jsonb)",
+                "INSERT INTO job (status, job_uuid, workflow, workflow_version, job_hash, ini) VALUES (?,?,?,?,?,?)",
```
Comment (SolomonShorser-OICR) [inline consonance-common/src/main/java/io/consonance/arch/persistence/PostgreSQL.java:192]: How/where are "extra_files" inserted now? Is it all in one transaction?

File: consonance-webservice/src/main/java/io/consonance/webservice/core/User.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        public User(String name) {
+                this.name = name;
+        }
+
+        public String getName() {
+                return name;
+        }
+
+        public int getId() {
+                return (int) (Math.random() * RANDOM_CONSTANT);
```
Comment (SolomonShorser-OICR) [inline consonance-webservice/src/main/java/io/consonance/webservice/core/User.java:21]: Why not just use the sequential ID from the database? Or a UUID from some UUID library?

File: consonance-webservice/src/main/java/io/consonance/webservice/jdbi/JobDAO.java [hunk 6]
```diff
@@ -30,15 +31,23 @@ public JobDAO(SessionFactory factory) {
...
     }
 
-    public long create(Job job) {
+    public int create(Job job) {
         return persist(job).getJobId();
     }
 
     public List<Job> findAll() {
-        return list(namedQuery("io.consonance.webservice.core.WorkflowRun.findAll"));
+        return list(namedQuery("io.consonance.arch.beans.core.Job.findAll"));
```
Comment (SolomonShorser-OICR) [inline consonance-webservice/src/main/java/io/consonance/webservice/jdbi/JobDAO.java:43]: Should the query name strings be stored in a constant somewhere?

File: consonance-common/src/main/java/io/consonance/arch/Base.java (hunk 3 above)
Comment (denis-yuen) [inline consonance-common/src/main/java/io/consonance/arch/Base.java:17]: Yep

File: consonance-common/src/main/java/io/consonance/arch/persistence/PostgreSQL.java (hunk 4 above)
Comment (denis-yuen) [inline consonance-common/src/main/java/io/consonance/arch/persistence/PostgreSQL.java:192]: The postgres class is now only used by the daemons which update. At some point, I'll clean up this code and remove their ability to insert. I think Hibernate should create the job and associated extra-files column in one transaction.

File: consonance-webservice/src/main/java/io/consonance/webservice/core/User.java (hunk 5 above)
Comment (denis-yuen) [inline consonance-webservice/src/main/java/io/consonance/webservice/core/User.java:21]: In the next pull request in the chain.

File: consonance-webservice/src/main/java/io/consonance/webservice/jdbi/JobDAO.java (hunk 6 above)
Comment (denis-yuen) [inline consonance-webservice/src/main/java/io/consonance/webservice/jdbi/JobDAO.java:43]: That's a good idea. Will do.

File: consonance-webservice/src/main/java/io/consonance/webservice/core/User.java (hunk 5 above)
Comment (SolomonShorser-OICR) [inline consonance-webservice/src/main/java/io/consonance/webservice/core/User.java:21]: Ok, I'll go have a look at that one.

**Pull request merged**